### PR TITLE
Update README to Improve Usability and Reproducibility of Code Base

### LINKED ISSUE #69: Improve Reproducibility and Reusability of Code Base
It is not straight-forward to run the code from the documentation provided. We need to extend and improve the documentation (starting with the README), and potentially, as a follow-up, also restructure the repository to make it easier for new users to find their way around, run the code and contribute to the code base.

### PR DESCRIPTION
**Pull request by MarionBWeinzierl**
Fixes #69 .

A number of changes to the README are made to make it easier to understand and use the code base.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,24 +8,23 @@ A Neural Net providing a subgrid parameterization of atmospheric convection in a
 The work is described in a [GRL paper](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2020GL091363) with [accompanying code available](https://github.com/yaniyuval/Neural_nework_parameterization/tree/v.1.0.3). The repository contains the neural net and its implementation into a simple system for atmospheric modelling, [SAM](http://rossby.msrc.sunysb.edu/~marat/SAM.html).
 
 The aims of this repository are to:
-1. develop a standalone fortran module based on this neural net that can be used elsewhere,
+1. provide a standalone fortran module of the convection parameterisation based on this neural net that can be used elsewhere,
 2. deploy the module in another atmospheric model, and
-3. evaluate its performance.
+3. evaluate its performance (and provide instructions/code to do this).
```
Comment (jatkinson1000) [inline README.md:13]: I guess this is now done in https://github.com/m2lines/CAM-ML.
Initially we were advised that all the work would be done here, but later it was suggested that we use a fork of CAM to implement it. So perhaps this needs changing now.
Though evaluation results might be reported on this repo once we finalise them.

File: README.md [hunk 2]
```diff
@@ -11,24 +8,23 @@ A Neural Net providing a subgrid parameterization of atmospheric convection in a
...
 
-We may also perform an investigation into interfacing the pytorch implementation of the Neural Net using the [pytorch-fortran bridging code](https://github.com/Cambridge-ICCS/fortran-pytorch-lib) developed at the [Institute of Computing for Climate Science](https://cambridge-iccs.github.io/).
+We have also started to perform an investigation into interfacing the pytorch implementation of the Neural Net using the [pytorch-fortran bridging code](https://github.com/Cambridge-ICCS/fortran-pytorch-lib) developed at the [Institute of Computing for Climate Science](https://cambridge-iccs.github.io/).
```
Comment (jatkinson1000) [inline README.md:15]: I wonder if we just remove this for now?
No longer a focus for M2LInES and we could add it back later if we do something?

File: README.md [hunk 3]
```diff
@@ -11,24 +8,23 @@ A Neural Net providing a subgrid parameterization of atmospheric convection in a
...
 
-The model will first be deployed into the [Single Column Atmospheric Model (SCAM)](https://www.cesm.ucar.edu/models/simple/scam) - a single column version of the CESM.
-We plan to evaluate performance using SCAM in the gateIII configuration for tropical convection in a similar manner described by the [SCAM6 pulication in JAMES](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2018MS001578).
-This will compare model performance to data from an intense observation period (IOP) described in an [AMS publication](https://journals.ametsoc.org/view/journals/atsc/36/1/1520-0469_1979_036_0053_saposs_2_0_co_2.xml).
+The model was first deployed into the [Single Column Atmospheric Model (SCAM)](https://www.cesm.ucar.edu/models/simple/scam) - a single column version of the CESM.
+We evaluate performance using Single Column CAM (SCAM) in the gateIII configuration for tropical convection in a similar manner described by the [SCAM6 pulication in JAMES](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2018MS001578).
```
Comment (jatkinson1000) [inline README.md:18]: More a note-to-self, we are going to move to focus on comparing to the TOGA II dataset which is a similar idea to GATE III run in SCAM. Documented in the same paper we already cite here.

File: README.md [hunk 4]
```diff
@@ -41,15 +37,68 @@ Long term developments of this project will seek to re-deploy more complex ML pa
...
+This can be done from within the `resources/` directory as follows:
+```
+python -m venv venv
+source venv/bin/activate
+pip install -r requirements.txt
+python sounding_to_netcdf.py
+deactivate
+
+```
+There are test routines associated with this code in `/tests/test_NN_module/`.
```
Comment (jatkinson1000) [inline README.md:66]:
```suggestion
There are test routines associated with this code in `/tests/test_YOG_convection/`.
```

File: README.md [hunk 5]
```diff
@@ -41,15 +37,68 @@ Long term developments of this project will seek to re-deploy more complex ML pa
...
+
+- `bld/namelist_files/namelist_definition.xml` - To add the new namelist variables.
+- `bld/namelist_files/namelist_defaults_cam.xml` - To set the new namelist variable defaults.
+- `/src/physics/cam/physcontrol.F90` - To read in the new namelist variables and broadcast them.
+- `/src/physics/cam/physpkg.F90` - To call the new parameterisation routines detailed in `yog_intr.F90`.
+
+Full details of these can be seen in the specific implementation in CAM linked above.
+
+There are test routines associated with this code in `/tests/test_CAM_interface/`.
+Guidance on running these can be found in the main README.
```
Comment (jatkinson1000) [inline README.md:99]:
```suggestion
Guidance on running these can be found below.
```

File: README.md (hunk 1 above)
Comment (MarionBWeinzierl) [reply README.md:13]: Do you have any suggestion on how to change the text here? Or leave it like this?

File: README.md (hunk 2 above)
Comment (MarionBWeinzierl) [reply README.md:15]: But there is still some leftover code/NN weights?

File: README.md (hunk 3 above)
Comment (MarionBWeinzierl) [reply README.md:18]: Do we want to mention it here?

Comment (jatkinson1000) [reply README.md:18]: Yes, I think maybe mention both GATE III and TOGA II?

File: README.md (hunk 2 above)
Comment (jatkinson1000) [reply README.md:15]: Not in the main repository that I can see, it's all in unmerged and stale (and messy) pull requests I think.

File: README.md (hunk 3 above)
Comment (MarionBWeinzierl) [reply README.md:18]: Ah, just seen that you said it's the same paper

Comment (MarionBWeinzierl) [reply README.md:18]: added that to the readme

File: README.md [hunk 6]
```diff
@@ -11,24 +8,23 @@ A Neural Net providing a subgrid parameterization of atmospheric convection in a
...
-
 ## Repository structure
 
 ```
 ├── YOG_convection
-│   └── ...
+│   └── ...
 ├── torch_nets
 │   └── ...
 ├── CAM_interface
```
Comment (jatkinson1000) [inline README.md:30]: We now have `profile_test` as of this morning.

File: README.md (hunk 1 above)
Comment (jatkinson1000) [reply README.md:13]: Not really...
I think leave for now perhaps.
I think the CAM-ML repo is linked from elsewhere in the README.

**Review by jatkinson1000 — COMMENTED** [review]
What is here now looks great to me, but you probably want to rebase on main and then bring over any extra readme content from profile_test/

File: README.md (hunk 6 above)
Comment (MarionBWeinzierl) [reply README.md:30]: added that

**Review by jatkinson1000 — APPROVED** [review]
Thanks @MarionBWeinzierl This is a real improvement to the readme and useability.

One thing might be to add a description of what is in `profile_test/`, but that is my fault so I could handle this elsewhere if you prefer so I'll approve now either way.

**Pull request merged**